bin/webpack-dev-server.js

```diff
@@ -294,7 +294,9 @@ function processOptions(webpackOptions) {
     };
   }
 
-  if (typeof options.stats === 'object' && typeof options.stats.colors === 'undefined') { options.stats.colors = argv.color; }
+  if (typeof options.stats === 'object' && typeof options.stats.colors === 'undefined') {
+    options.stats = Object.assign({}, options.stats, { colors: argv.color });
```

Comment:
I may be having a brain fart here, so please correct me if I'm wrong. this doesn't look like it'll avoid mutating the `stats` property. there's a larger issue here (which is resolved in the v3 beta branch) of the options passed being mutated on the whole:

```
var a = { stats: {} };

function doit (o) {
  o.stats = Object.assign({}, a.stats, { color: '0' });
}

doit(a);
a;

// returns
// {stats: {…}}
//   stats: {color: "0"}
//   __proto__: Object
```

I could be missing something, but I think the solution is to prevent the top level options object passed to devServer from being mutated.

Reply:
You are right, but it's the `stats` property *value* that shouldn't be mutated.

```
var stats = {};
var a = { stats: stats };

function doit (o) {
  o.stats = Object.assign({}, a.stats, { color: '0' });
}

doit(a);
stats;

// {}
```

Reply:
I see, I see. Hadn't considered that case. Could you put a simple test for this in `/test`?

Reply:
Sadly I cannot write a good test case because when running `test/cli.test.js`, the package `supports-color` detect that it doesn't run in a tty, so it returns `false` (which webpack sees as valid).